### DISCUSSION TITLE
HelmRelease timeout override

### DISCUSSION THREAD

**Discussion by DevopsCareer** [fluxcd/flux2#4879] [General]
Hello, 
I have a HelmRelease.spec.timeout set as 30 mins for my HelmCharts. It works ifine n many cases,but in few cases we notice that due to bad configuration push the HelmRelease fails and the dev/user have to wait for 30 minutes before we push fixed configuration through flux-cd reconciliation. 
What we want is an override option,  so in this scenario if I realized my bad configuration within in say 10 mins instead of waiting for 30 min counter to expire -  I would like HelmRelease to stop and not wait for 30 min timer to expire and immediately push new configuration through pipeline. Any manual intervention , cmd line is also fine for cases we realize issue soon.

We have kept upper limit of 30 mins because this is max time we need for our pods to to roll forward of config are good, so reducing that time is not an option.

**Comment by stefanprodan** [comment]
If you have access to the cluster, restarting the helm-controller pod will do.

**Reply by DevopsCareer** [reply]
Yes, I have access to cluster let me test this. I am assuming the restart of Helm controller pod will not have any effect on already running (previous release) pods/containers which helm controller managed through helm release.

**Reply by stefanprodan** [reply]
If you make a change to the HelmRelease, when you restart the controller, it will take into account the new desired state.

**Reply by DevopsCareer** [reply]
so :
1. Make a HelmRelease change for desired service
2.Restart Helm controller pod?
Is there a direct command just to restart Helm controller?

**Reply by stefanprodan** [reply]
kubectl -n flux-system rollout restart deploy/helm-controller --field-manager flux-client-side-apply

**Reply by DevopsCareer** [reply]
Thanks let me try this.
